crates/ruff/src/rules/pep8_naming/rules/mixed_case_variable_in_class_scope.rs

```diff
@@ -36,6 +37,14 @@ pub fn mixed_case_variable_in_class_scope(
     {
         return;
     }
+    if bases.iter().any(|base| {
+        if let ExprKind::Name { id, .. } = &base.node {
+            return id == "TypedDict";
+        }
+        false
+    }) {
+        return;
+    }
     if helpers::is_mixed_case(name) && !helpers::is_namedtuple_assignment(checker, stmt) {
```

Comment:
Let's maybe move to a helper (`is_typeddict`) and do the check at the end here (`&& !helpers...`).